model/src/clients/testsys_client.rs

```diff
@@ -0,0 +1,1301 @@
+                failed: None,
+            });
+        }
+        for crd in &self.crds {
+            results.extend(crd.create_result_row());
+        }
+        results.sort_by(|a, b| a.name.cmp(&b.name));
+
+        Table::new(results)
+            .with(Style::blank())
+            .with(Modify::new(Full).with(Alignment::left()))
+    }
+
+    pub fn to_string(&self, width: usize) -> String {
+        self.create_table()
+            .with(MaxWidth::truncating(width))
+            .with(MinWidth::new(width))
+            .to_string()
+    }
+}
+
+#[derive(Tabled, Default, Clone, Serialize)]
+struct ResultRow {
+    #[tabled(rename = "NAME")]
+    name: String,
+    #[tabled(rename = "TYPE")]
+    object_type: String,
+    #[tabled(rename = "STATE")]
+    state: String,
+    #[tabled(rename = "PASSED")]
+    #[tabled(display_with = "display_option")]
+    passed: Option<u64>,
+    #[tabled(rename = "SKIPPED")]
+    #[tabled(display_with = "display_option")]
+    skipped: Option<u64>,
+    #[tabled(rename = "FAILED")]
+    #[tabled(display_with = "display_option")]
+    failed: Option<u64>,
+}
+
+fn display_option(o: &Option<u64>) -> String {
+    match o {
+        Some(count) => format!("{}", count),
+        None => "".to_string(),
+    }
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// The error type for `TestsysClient`
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Unable to {}: {}", action, source))]
+    Client {
+        action: String,
+        source: crate::clients::Error,
+    },
+
+    #[snafu(display("Unable to create client: {}", source))]
+    ClientCreateKubeconfig {
+        source: kube::config::KubeconfigError,
+    },
+
+    #[snafu(display("Unable to read kubeconfig: {}", source))]
+    ConfigRead {
+        source: kube::config::KubeconfigError,
+    },
+
+    #[snafu(display("Error Creating {}: {}", what, source))]
+    Create { what: String, source: kube::Error },
+
+    #[snafu(display("Unable to send delete event"))]
+    DeleteEvent,
+
+    #[snafu(display("Unable to read file '{}': {}", path.display(), source))]
+    File {
+        path: PathBuf,
+        source: std::io::Error,
+    },
+
+    #[snafu(display("Unable to {}: {}", action, source))]
+    Io {
+        action: String,
+        source: std::io::Error,
+    },
+
+    #[snafu(display("Unable to {}: {}", action, source))]
+    Kube { action: String, source: kube::Error },
+
+    #[snafu(display("Could not serialize object: {}", source))]
+    JsonSerialize { source: serde_json::Error },
+
+    #[snafu(display("Unable to find {}", what))]
+    NotFound { what: String },
+
+    #[snafu(display("Unable to send event: {}", source))]
+    Sender {
+        source: futures::channel::mpsc::SendError,
+    },
+
+    #[snafu(display("Unable to {}: {}", action, source))]
+    SerdeYaml {
+        action: String,
+        source: serde_yaml::Error,
+    },
+}
+
+#[derive(Serialize)]
+pub(crate) struct DockerConfigJson {
```

Comment:
Small types like this I might suggest putting in the `mod.rs` (not `client/mod.rs` though, I think there needs to be a sibling directory, e.g. `utils/mod.rs`)